File: fix_debug.py
import re

with open('backend/routes/scans.js', 'r', encoding='utf-8') as f:
    routes = f.read()

debug_route = '''// Temporary debug route to list models
router.get('/debug-models', async (req, res) => {
  try {
    const response = await fetch("https://generativelanguage.googleapis.com/v1beta/models?key=" + process.env.GEMINI_API_KEY);
    const data = await response.json();
    res.json(data);
  } catch (error) {
    res.status(500).json({ error: error.message });
  }
});
'''

# Remove the old debug route at the bottom
routes = re.sub(r'// Temporary debug route to list models[\s\S]*?\}\);', '', routes)

# Insert it at the top, just after router initialization
routes = routes.replace('const router = express.Router();', 'const router = express.Router();\n\n' + debug_route)

with open('backend/routes/scans.js', 'w', encoding='utf-8') as f:
    f.write(routes)

File: fix_debug.js
const fs = require('fs');
let code = fs.readFileSync('backend/routes/scans.js', 'utf8');

code = code.replace(/https:\/\/generativelanguage\.googleapis\.com\/v1beta\/models\?key="\s*\+\s*process\.env\.GEMINI_API_KEY/g, `"https://api.groq.com/openai/v1/models", { headers: { "Authorization": "Bearer " + process.env.GROQ_API_KEY } }`);

fs.writeFileSync('backend/routes/scans.js', code);
console.log("DEBUG FIXED");
